src/ceph-volume/ceph_volume/process.py

```diff
@@ -99,22 +99,19 @@ def run(command, **kw):
     """
     stop_on_error = kw.pop('stop_on_error', True)
     command_msg = obfuscate(command, kw.pop('obfuscate', None))
-    stdin = kw.pop('stdin', None)
+    assert 'stdin' not in kw, "'stdin' not supported by process.run()"
```

Comment:
Please don't use `assert` like this in Python. Since there is no interface that calls `process.run` with `stdin` it is safe to remove.

Reply:
i am fine removing this. but in general, i think a use case of  `assert` in Python is to avoid possible caller passing undesired parameter.

> Since there is no interface that calls `process.run` with `stdin` it is safe to remove.

that's exactly why i want to add an assert here: to avoid future interfaces calling `process.run` with `stdin`, if any, instead of taking times to debug it, we can fail early, and have a nice backtrace.

Reply:
Two important reasons on why we don't use `assert` in Python *to control program flow* are: 

* Plain asserts can be removed by Python when environment variables or flags are used (see `python -O` or `PYTHONOPTIMIZE`)
* It doesn't offer granularity as to what exceptions one should be catching. In the (unlikely) situation some other tool is using plain assertions as a generic stop-all, one cannot safely catch those and ignore some others (and vice versa)

These interfaces are very well documented, and are clear in what they support. The usage of keyword arguments (`kw`) makes it lenient to accept anything and everything, but if adding a check for one variable, then checks must be implemented for every variable that you think may cause an error because it is not being used (whitelist/blacklist). It would also need to be added to `process.call` since the same `kw` mechanism is being used there. We haven't added such a mechanism because we are betting on well-documented functions.

I am -1 on adding a check for `process.run` even with granular exception for this because it then must implement something for all the other options it must not accept via `kw`.

Reply:
faire enough. but if it's documented that `input` is not a valid parameter, i would not have used it. anyway, i will remove it.

Reply:
also, i am not using `assert` for `control program flow`. it servers as way for debugging and documentation.